packages/gatsby-source-contentful/src/gatsby-node.js

```diff
@@ -23,7 +23,118 @@ const restrictedNodeFields = [
 
 exports.setFieldsOnGraphQLNodeType = require(`./extend-node-type`).extendNodeType
 
-exports.onPreBootstrap = validateOptions
+const validateContentfulAccess = async pluginOptions => {
+  if (process.env.NODE_ENV === `test`) return undefined
+
+  await fetch(
+    `https://${pluginOptions.host}/content/v1/spaces/${pluginOptions.spaceId}`,
+    {
+      headers: {
+        Authorization: `Bearer ${pluginOptions.accessToken}`,
+        "Content-Type": `application/json`,
+      },
+    }
+  )
+    .then(res => res.json())
+    .then(json => {
+      if (json.errors)
+        throw new Error(
+          `Cannot access Contentful Space ${maskText(pluginOptions.spaceId)}`
+        )
+    })
+    .catch(err => {
+      throw new Error(`Fetch call to check Contentful access failed.`)
+    })
+
+  return undefined
```

Comment:
I assume this is because there's an early return, but since this isn't a .ts file, I'm not sure?

Reply:
For external fro Joi API doc https://github.com/sideway/joi/blob/master/API.md#anyexternalmethod-description

`which can either return a replacement value, undefined to indicate no change, or throw an error.`